corehq/apps/case_importer/suggested_fields.py

```diff
@@ -0,0 +1,88 @@
+        if field not in exclude_fields and field not in combined_field_specs:
+            combined_field_specs[field] = field_spec
+
+    return sorted(combined_field_specs.values(), key=lambda field_spec: field_spec.field)
+
+
+def get_suggested_case_fields(domain, case_type, exclude=None):
+    exclude_fields = set(RESERVED_FIELDS) | set(exclude or [])
+
+    special_field_specs = (field_spec for field_spec in get_special_fields())
+
+    dynamic_field_specs = (FieldSpec(field=field, show_in_menu=True)
+                           for field in get_case_properties_for_case_type(domain, case_type))
+
+    return _combine_field_specs(
+        itertools.chain(special_field_specs, dynamic_field_specs),
+        exclude_fields=exclude_fields
+    )
+
+
+class FieldSpec(jsonobject.StrictJsonObject):
+    field = jsonobject.StringProperty()
+    description = jsonobject.StringProperty()
+    show_in_menu = jsonobject.BooleanProperty(default=False)
+    discoverable = jsonobject.BooleanProperty(default=True)
+
+
+def get_special_fields():
+    return [
+        FieldSpec(
+            field='name',
+            description=_("This field will be used to set the case's name."),
+            show_in_menu=True),
+        FieldSpec(
+            field='owner_name',
+            description=_("This field will assign the case to a new owner given by "
+                          "Username, Group name, or Organization name."),
+        ),
+        FieldSpec(
+            field='owner_id',
+            description=_("This field will assign the case to a new owner given by "
+                          "User ID, Group ID, or Organization ID.")),
+        FieldSpec(
+            field='external_id',
+            description=_("This field will set the case's external_id")),
+        FieldSpec(
+            field='parent_external_id',
+            description=_("This field will assign the case a new parent given by "
+                          "the parent case's external_id. "
+                          "You must use along with parent_type.")),
+        FieldSpec(
+            field='parent_id',
+            description=_("This field will assign the case a new parent given by "
+                          "the parent's Case ID. "
+                          "You must use along with parent_type.")),
+        FieldSpec(
+            field='parent_type',
+            description=_("Use to specify the parent's case type. "
+                          "Usually used with parent_id or parent_external_id")),
+        FieldSpec(
+            field='parent_ref',
+            description=_("This is a deprecated feature needed for a handful of clients. "
+                          "Please do not use unless you know what you are doing"),
+            discoverable=False),
+        FieldSpec(
+            field='close',
+            description=_("This field will be used to close cases. "
+                          "Any case with 'yes' in this column will be closed.")),
+    ]
```

Comment:
Yeah, I started this whole thing by just writing out these specs for myself, before even having the feature for showing them to the user. It just felt kind of crazy that there were like 5 different places where special magic words were used in `case_importer` and no place that defined them as special (or bothered to mention this to the user!).